gpii/node_modules/flowManager/src/PSPChannel.js

```diff
@@ -300,3 +323,26 @@ gpii.pspChannel.receiveMessage = function (message, pspChannel) {
     }
     pspChannel.applier.change(message.path, message.value, "ADD", "PSP");
 };
+
+gpii.pspChannel.generateDefaultSettingControls = function (defaultSettings, schemas) {
+    var defaultSettingControls = {};
+    var defaultPreferences = fluid.get(defaultSettings, ["contexts", "gpii-default", "preferences"]);
+    fluid.each(defaultPreferences, function (prefsVal, prefsKey) {
+        gpii.pspChannel.emitSettingControl(defaultSettingControls, schemas[prefsKey], prefsVal, [prefsKey], "live");
+    });
+    return defaultSettingControls;
+};
+
+gpii.pspChannel.respondDefaultSettings = function (model, transformSpec) {
+    var defaultSettings = transformSpec.defaultSettings;
+    if (!model || !defaultSettings) {
+        return model;
+    } else {
+        if ($.isEmptyObject(model.settingControls)) {
```

Comment:
I think this is too narrow a mission for the function - it should report the default settings for *any* control which does not appear, which has an entry in the default settings. Because we already implemented https://issues.gpii.net/browse/GPII-3434 , morally we consider that any keyin action has its effect combined with a "reset to standard".
In practice this will leave us with an even simpler implementation than the one we have here.

Reply:
Once the initial empty settingControls block is replaced by default settings. These settings will remain in the model and be merged with following settings requested by PSP clients.

If you compare [the new test for this JIRA](https://github.com/cindyli/universal/blob/GPII-3828/tests/shared/PSPIntegrationTestDefs.js#L806-L826) with [its counterpart test above](https://github.com/cindyli/universal/blob/GPII-3828/tests/shared/PSPIntegrationTestDefs.js#L743-L758) that has no default settings, all following settingControls block responded to PSP clients has default settings combined.

Let me know if you see problems with this handling. Thanks.

Reply:
I think perhaps this is the problem - we should not be leaving any settings in the model, but instead we should be updating the "default" value in the reported schema for the setting as I've clarified in the JIRA comment. This should *always* be the default schema value reported, whether there is any setting value or not.